### PR TITLE
Fix padding of zeros in keys of quasi-distribution and probabilities

### PR DESCRIPTION
**Pull request by ajavadia**
previously there would be keys like "0" and "10". Now they are "00" and "10" which is consistent with how the Counts are done.

### CODE REVIEW

File: releasenotes/notes/fix-quasi-probabilty-zero-padding-1a09a41eecb19b12.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+fixes:
```
Comment (mtreinish) [inline releasenotes/notes/fix-quasi-probabilty-zero-padding-1a09a41eecb19b12.yaml:2]: You don't need a release note this hasn't been released yet so it's covered by the new feature release note

File: qiskit/result/distributions/quasi.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 class QuasiDistribution(dict):
     """A dict-like class for representing qasi-probabilities."""
 
-    bitstring_regex = re.compile(r"^[01]+$")
+    _bitstring_regex = re.compile(r"^[01]+$")
```
Comment (mtreinish) [inline qiskit/result/distributions/quasi.py:29]: Why change this here and not in Probabilities?

File: qiskit/result/distributions/quasi.py [hunk 3]
```diff
@@ -38,8 +38,7 @@ def __init__(self, data, shots=None):
                 The keys can be one of several formats:
 
                     * A hexadecimal string of the form ``"0x4a"``
-                    * A bit string prefixed with ``0b`` for example
-                      ``'0b1011'``
+                    * A bit string e.g. ``'0b1011'`` or ``"01011"``
```
Comment (mtreinish) [inline qiskit/result/distributions/quasi.py:41]: Do you want to update the docstring on Probabilities too? It behaves the same way.

**Review by mtreinish — COMMENTED** [review]
LGTM, just some small nits inline on making things consistent between the 2 classes

**Comment by ajavadia** [comment]
@mtreinish updated according to your comments

**Review by mtreinish — APPROVED** [review]
LGTM, thanks for the quick update

**Pull request merged**